agentops/helpers.py

```diff
@@ -99,22 +106,7 @@ def remove_unwanted_items(value):
 
 
 def check_call_stack_for_agent_id() -> Union[UUID, None]:
-    for frame_info in inspect.stack():
-        # Look through the call stack for the class that called the LLM
-        local_vars = frame_info.frame.f_locals
-        for var in local_vars.values():
-            # We stop looking up the stack at main because after that we see global variables
-            if var == "__main__":
-                return None
-            if hasattr(var, "agent_ops_agent_id") and getattr(
-                var, "agent_ops_agent_id"
-            ):
-                logger.debug(
-                    "LLM call from agent named: %s",
-                    getattr(var, "agent_ops_agent_name"),
-                )
-                return getattr(var, "agent_ops_agent_id")
-    return None
+    return AgentOpsDescriptor.from_stack()
```

Comment:
Everywhere in this PR we instantiate `AgentOpsDescriptor` inline instead of having class instance variables. Why are we doing this instead of just running `from_stack` as a separately defined helper function?

Reply:
The signature would've better been off `AgentOpsDescriptor.get_from_stack(property="agent_id")`

> The reason you want a descriptor is because you need a distinguishable way of identifying your own properties (agentops props)

> Everywhere in this PR we instantiate AgentOpsDescriptor inline instead

...it's not being instantiated, it's a static method being invoked, as if it was a function under a module named `AgentOpsDescriptor`. 

Most of the "why" are answered with behavior encapsulation